src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5342,35 +5360,39 @@ private static List<CompletionResult> CompleteRequires(CompletionContext context
                 {
                     if (moduleSpecKey.StartsWith(currentValue, StringComparison.OrdinalIgnoreCase))
                     {
-                        results.Add(new CompletionResult(moduleSpecKey, moduleSpecKey, CompletionResultType.ParameterValue, s_requiresModuleSpecKeys[moduleSpecKey]));
+                        string toolTip = ResourceManagerCache.GetResourceString(
+                        typeof(CompletionCompleters).Assembly,
+                        "System.Management.Automation.resources.TabCompletionStrings",
+                        $"RequiresModuleSpec{moduleSpecKey}Description");
+                        results.Add(new CompletionResult(moduleSpecKey, moduleSpecKey, CompletionResultType.ParameterValue, toolTip));
                     }
                 }
             }
 
             return results;
         }
 
-        private static readonly IReadOnlyDictionary<string, string> s_requiresParameters = new SortedList<string, string>(StringComparer.OrdinalIgnoreCase)
+        private static readonly string[] s_requiresParameters = new string[]
         {
-            { "Modules", "Specifies PowerShell modules that the script requires." },
-            { "PSEdition", "Specifies a PowerShell edition that the script requires." },
-            { "RunAsAdministrator", "Specifies that PowerShell must be running as administrator on Windows." },
-            { "Version", "Specifies the minimum version of PowerShell that the script requires." },
+            "Modules",
+            "PSEdition",
+            "RunAsAdministrator",
+            "Version"
         };
 
-        private static readonly IReadOnlyDictionary<string, string> s_requiresPSEditions = new SortedList<string, string>(StringComparer.OrdinalIgnoreCase)
+        private static readonly string[] s_requiresPSEditions = new string[]
         {
-            { "Core", "Specifies that the script requires PowerShell Core to run." },
-            { "Desktop", "Specifies that the script requires Windows PowerShell to run." },
+            "Core",
+            "Desktop"
         };
 
-        private static readonly IReadOnlyDictionary<string, string> s_requiresModuleSpecKeys = new SortedList<string, string>(StringComparer.OrdinalIgnoreCase)
+        private static readonly string[] s_requiresModuleSpecKeys = new string[]
         {
-            { "ModuleName", "Required. Specifies the module name." },
-            { "GUID", "Optional. Specifies the GUID of the module." },
-            { "ModuleVersion", "Specifies a minimum acceptable version of the module." },
-            { "RequiredVersion", "Specifies an exact, required version of the module." },
-            { "MaximumVersion", "Specifies the maximum acceptable version of the module." },
+            "ModuleName",
+            "GUID",
+            "ModuleVersion",
+            "RequiredVersion",
+            "MaximumVersion"
```

Comment:
I think it makes sense to sort the list here. (I see you had to change a test.)

Reply:
It is sorted the way I intended them to be sorted when I originally added them. Instead of sorting them alphabetically I sort them by significance/my own preference. If I see a module specification hashtable I think it makes most sense to have the module name as the first element followed by the the GUID and the specified version.

Reply:
With Ctrl-Space users get full list on screen and I'd prefer to see elements in ABC order.

Reply:
There are other places where things aren't sorted alphabetically such as command parameters which are sorted by the order in which they were defined followed by the common parameters.  
Windows Explorer also sorts the view so folders are grouped together at the top when sorting by name and TBH I wish PS would do the same when tab completing files.  
I'm interested in other opinions on this. I'll change it if I'm all alone about this opinion but I think it's good to have the most important suggestions like "ModuleName" or "SYNOPSIS" at the top when completing module specifications and help.

Reply:
Common parameters are a separate history. (For example there was a suggestion to show them only by a special combination of keys.)

>  command parameters which are sorted by the order in which they were defined

I don't think anyone thinks about that when they add a new parameter. :-)

For cmdlets that have 10 or more parameters like Select-String, this is really annoying. Now that we always have Predictor on, it's not so critical. And as a result we can now be consistent and have all these lists sorted so that a full screen script would be handy. The only exception would be to put the last used value first, but Predictor already does that.